src/components/ActionMenu/ActionMenu.tsx

```diff
@@ -23,48 +23,28 @@ export interface ActionMenuProps {
   rollup?: boolean;
 }
 
-interface State {
-  activeMenuGroup?: string;
-}
-
-export class ActionMenu extends PureComponent<ActionMenuProps, State> {
-  static contextType = FeaturesContext;
-  context!: React.ContextType<typeof FeaturesContext>;
-
-  state: State = {
-    activeMenuGroup: undefined,
-  };
-
-  render() {
-    const {actions = [], groups = [], rollup} = this.props;
-
-    if (actions.length === 0 && groups.length === 0) {
-      return null;
-    }
+export function ActionMenu({
+  actions = [],
+  groups = [],
+  rollup,
+}: ActionMenuProps) {
+  const {newDesignLanguage} = useFeatures();
+  const [activeMenuGroup, setActiveMenuGroup] = useState<string | undefined>(
+    undefined,
+  );
+
+  if (actions.length === 0 && groups.length === 0) {
+    return null;
+  }
 
-    const actionMenuClassNames = classNames(
-      styles.ActionMenu,
-      rollup && styles.rollup,
-    );
+  const actionMenuClassNames = classNames(
+    styles.ActionMenu,
+    rollup && styles.rollup,
+  );
 
-    const rollupSections = groups.map((group) => convertGroupToSection(group));
+  const rollupSections = groups.map((group) => convertGroupToSection(group));
 
-    return (
-      <div className={actionMenuClassNames}>
-        {rollup ? (
-          <RollupActions items={actions} sections={rollupSections} />
-        ) : (
-          this.renderActions()
-        )}
-      </div>
-    );
-  }
-
-  // eslint-disable-next-line @shopify/react-no-multiple-render-methods
-  private renderActions = () => {
-    const {newDesignLanguage} = this.context || {};
-    const {actions = [], groups = []} = this.props;
-    const {activeMenuGroup} = this.state;
+  function renderActions() {
```

Comment:
Extract this out so it - A definition of a react component inside another react component is odd, and probably bad for performance as it means the function gets recreated on every render.

Could do `function Actions() {}`, then below you can replace `renderActions()` with `<Actions/>` (possibly with some props needed)